tfhe/src/core_crypto/fft_impl/crypto/bootstrap.rs

```diff
@@ -320,3 +321,114 @@ pub fn pbs_modulus_switch<Scalar: UnsignedTorus + CastInto<usize>>(
     output <<= lut_count_log.0;
     <Scalar as CastInto<usize>>::cast_into(output)
 }
+
+pub trait FourierBootstrapKey<Scalar> {
+    type Fft;
+
+    fn new_fft(polynomial_size: PolynomialSize) -> Self::Fft;
+
+    fn new(
+        input_lwe_dimension: LweDimension,
+        polynomial_size: PolynomialSize,
+        glwe_size: GlweSize,
+        decomposition_base_log: DecompositionBaseLog,
+        decomposition_level_count: DecompositionLevelCount,
+    ) -> Self;
+
+    fn fill_with_forward_fourier<ContBsk>(
+        &mut self,
+        coef_bsk: &LweBootstrapKey<ContBsk>,
+        fft: &Self::Fft,
+    ) where
+        ContBsk: Container<Element = Scalar>;
+
+    fn bootstrap_scratch(
+        glwe_size: GlweSize,
+        polynomial_size: PolynomialSize,
+        fft: &Self::Fft,
+    ) -> Result<StackReq, SizeOverflow>;
+
+    fn bootstrap<ContLweOut, ContLweIn, ContAcc>(
+        &self,
+        lwe_out: &mut LweCiphertext<ContLweOut>,
+        lwe_in: &LweCiphertext<ContLweIn>,
+        accumulator: &GlweCiphertext<ContAcc>,
+        fft: &Self::Fft,
+        stack: PodStack<'_>,
+    ) where
+        ContLweOut: ContainerMut<Element = Scalar>,
+        ContLweIn: Container<Element = Scalar>,
+        ContAcc: Container<Element = Scalar>;
+}
+
+impl<Scalar> FourierBootstrapKey<Scalar> for FourierLweBootstrapKeyOwned
+where
+    Scalar: UnsignedTorus + CastInto<usize>,
+{
+    type Fft = Fft;
+
+    fn new_fft(polynomial_size: PolynomialSize) -> Self::Fft {
+        Fft::new(polynomial_size)
+    }
+
+    fn new(
+        input_lwe_dimension: LweDimension,
+        polynomial_size: PolynomialSize,
+        glwe_size: GlweSize,
+        decomposition_base_log: DecompositionBaseLog,
+        decomposition_level_count: DecompositionLevelCount,
+    ) -> Self {
+        Self::new(
+            input_lwe_dimension,
+            glwe_size,
+            polynomial_size,
+            decomposition_base_log,
+            decomposition_level_count,
+        )
+    }
+
+    fn fill_with_forward_fourier<ContBsk>(
+        &mut self,
+        coef_bsk: &LweBootstrapKey<ContBsk>,
+        fft: &Self::Fft,
+    ) where
+        ContBsk: Container<Element = Scalar>,
+    {
+        let fft = fft.as_view();
+        let mut mem = GlobalPodBuffer::new(fill_with_forward_fourier_scratch(fft).unwrap());
```

Comment:
any reason to take a GlobalPodBuffer here rather than a PodStack directly ?

Reply:
i don't take a GlobalPodBuffer as a parameter, i create it inside the function impl
i figured it might not be that bad to allocate in `fill_with_forward_fourier` directly since the allocation is small (about [f64; 512]), and only the f64 fft needs to allocate during the conversion

Reply:
sorry, I meant that other primitives require a podstack (your original bootstrap implem for converting the key), but yes generally for the one time conversion it's ok, was just wondering why you preferred the Global buffer this time

Reply:
i just forgot to add the podstack initially and went along with it :sweat_smile:

Reply:
ok your call ! I think I would keep the podstack as an input for consistency and to allow higher levels to have more control on that if that's ok 🙂